internal/auth/oidc/testing.go

```diff
@@ -237,3 +237,16 @@ func testProvider(t *testing.T, clientId, clientSecret, allowedRedirectURL strin
 	require.NoError(err)
 	return p1
 }
+
+// testFreePort just returns an available free localhost port
+func testFreePort(t *testing.T) int {
```

Comment:
This is actually a great helper function -- can you put it in `internal/testing` so it's more broadly available?

Reply:
sounds good.  I'll open a PR after this to move it under the sdk, and once it's merged, I'll open an oidc PR to start using it here and remove it from this package.